polaris/curation/_data_curator.py

```diff
@@ -195,49 +208,71 @@ def check_outliers(data, data_cols, method: str = "zscore", prefix="OUTLIER", **
 def run_data_curation(
     data: pd.DataFrame,
     data_cols: List[str],
+    data_types: List[str] = None,
     mask_stereo_undefined_mols: bool = False,
     ignore_stereo: bool = False,
     class_thresholds: Dict = None,
     outlier_params: Dict = None,
     activity_cliff_params: Dict = None,
+    keep_all_rows: bool = True,
 ) -> pd.DataFrame:
     """Perform curation on the measured values from biological assay
 
     Args:
         data: Dataframe which include molecule, measured values.
         data_cols: Column names for the data of interest.
+        data_types: Data type of the above data columns. The types will be detected not provided.
         mask_stereo_undefined_mols: Whether mask the molecule stereo pairs which show activity cliff from dataset.
         ignore_stereo: Ignore the stereochemistry information from data curation.
         class_thresholds: Parameters to define class labels for the above listed `data_cols`.
         outlier_params: Parameters for outlier detection.
         activity_cliff_params: Parameter for identify the activity cliff between the stereoisomers.
-
-    See Also:
-
+        keep_all_rows: Whether skip merging the replicated molecules. It's recommend to set to True during data audit process.
     """
-    data = check_outliers(data, data_cols, **outlier_params if outlier_params is not None else {})
+    if data_types is None:
+        logger.warning("Data types are not provides. We will try our best to determine the types")
+        data_types = [
+            type_of_target(data[data_col].dropna().values, input_name=data_col) for data_col in data_cols
+        ]
+
+    data_cols_reg = [
+        data_col for data_col, data_type in zip(data_cols, data_types) if data_type in CONTINUOUS
+    ]
+
+    data = check_outliers(
+        data,
+        data_cols,
+        data_types,
+        **outlier_params if outlier_params is not None else {},
+    )
 
     # User should deal with scaling and normalization on their end
-    data = _merge_duplicates(
-        data=data,
-        data_cols=data_cols,
-        merge_on=[NO_STEREO_UNIQUE_ID] if ignore_stereo else [UNIQUE_ID],
-    )
-    # class conversion
+    # The merging should only be performed for continuous data
```

Comment:
Why only for continuous? Is it because we don't know how to merge a data point with conflicting classes?